### PR TITLE
Sortition Tree Tests

### PR DESCRIPTION
**Pull request by beaushinkle**
This PR overhauls the sortition tree tests!

Summary of changes:

+ Simplified the test layout (removed unnecessary `beforeEach` constructions)
+ Simplified the `position` constructions to make it more human-readable
+ Heavily annotated the tests
+ Added some additional specs where the assertions seemed sparse

It might be useful to review commit by commit

Tagging @eth-r @lukasz-zimnoch @pdyraga

### CODE REVIEW

File: test/sortitionTreeTest.js [hunk 1]
```diff
@@ -19,86 +19,122 @@ describe("SortitionTree", () => {
...
     context("when one leaf is set", () => {
-      beforeEach(async () => {
-        const weight1 = 0x1234
-        const position1 = parseInt("00123456", 8)
-
-        const leaf = await sortition.toLeaf(alice.address, weight1)
-        await sortition.publicSetLeaf(position1, leaf, weight1)
-      })
-
       it("should return correct value for the tree", async () => {
```
Comment (pdyraga) [inline test/sortitionTreeTest.js:22]: ... fot the tree with a leaf in the first slot?

File: test/sortitionTreeTest.js [hunk 2]
```diff
@@ -19,86 +19,122 @@ describe("SortitionTree", () => {
...
-
-      it("should return correct value for the tree", async () => {
         const root = await sortition.getRoot()
-        expect(ethers.utils.hexlify(root)).to.be.equal("0x1000fff1")
+        expect(ethers.utils.hexlify(root)).to.be.equal("0x1000fff1") // weightA + weightB
+        // The full output here looks like
+        // 0x00000000,00000000,00000000,00000000,00000000,00000000,00000000,1000fff1
+        //  slot 7     slot 6   slot 5   slot 4   slot 3   slot 2   slot 1   slot 0
+        // without the commas added for readability. All the padding zeros are
+        // dropped when we hexlify, which simplifies to 0x1100000000.
```
Comment (pdyraga) [inline test/sortitionTreeTest.js:131]: It simplifies to `0x1000fff1`

**Review by pdyraga — APPROVED** [review]
Two comments, will be addressed in a followup PR.

File: test/sortitionTreeTest.js (hunk 2 above)
Comment (beaushinkle) [reply test/sortitionTreeTest.js:131]: right you are sir

Comment (beaushinkle) [reply test/sortitionTreeTest.js:131]: [6241b8e](https://github.com/keep-network/sortition-pools/pull/180/commits/6241b8e5fbf9d1878c92ae2b71c60a2393d736d8)

File: test/sortitionTreeTest.js (hunk 1 above)
Comment (beaushinkle) [reply test/sortitionTreeTest.js:22]: [415e151](https://github.com/keep-network/sortition-pools/pull/180/commits/415e151be37db3bd0ae3202a20c29092ff7ff61f)

**Pull request merged**